src/transcode/interface.ts

```diff
@@ -0,0 +1,28 @@
+import { EventFragment, Interface, Result } from 'ethers/lib/utils';
+import { argType, EventItem, join248bitChunks, splitInto248BitChunks } from '../utils/event';
+
+export class WarpInterface extends Interface {
+  decodeWarpEvent(fragment: EventFragment | string, warpEvent: EventItem): Result {
```

Comment:
There are no tests using `decodeWarpEvent`, which is what the hardhat integration and other tools will make most use of.
Can you add a few tests for it?

Reply:
tests similar to test/encode.test.ts would be great

Reply:
on it, thanks